packages/react-instantsearch/examples/next-app/pages/index.js

```diff
@@ -26,7 +26,7 @@ export default class extends React.Component {
      to initialize the searchState. 
   */
   static async getInitialProps(params) {
-    const searchState = qs.parse(params.asPath.substring(2));
+    const searchState = qs.parse(asPath.substring(asPath.indexOf('?') + 1))
```

Comment:
Oh Sorry! should be `params.asPath`

Also, I am doing custom routing so my asPath is different from href. 

asPath can be `/deals/store/amazon?query=1` 
`.substring` would fail in this case.